### PR TITLE
Shaoqz/refactor mutation api

### PR DESCRIPTION
**Pull request by shaoqx**
- Mainly finished `mutation_pattern` submodule that designs a pattern to specify mutants.
- Fixed many bugs along the way.
- Created the backbone for the new mutation module, still need to finishes several functions.
- Support pymol2 package in `PyMolInterface`
- Add structure_selection submodule for selecting part of the structure using pymol-based pattern
- Add a template for docstring of Science APIs

### CODE REVIEW

File: enzy_htp/chemical/atoms.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 X_H_BOND_LENGTH: Dict[str, float] = {
         "C": 1.07,
         "N": 1.0
-}
+} # yapf: disable
```
Comment (chrisjurich) [inline enzy_htp/chemical/atoms.py:12]: Get rid of this comment.

File: enzy_htp/core/general.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    """convert the result of get_interval_from_list() to a str"""
+    result = get_interval_from_list(target_list)
+    result = ",".join([f"{x[0]}-{x[1]}" if x[0] != x[1] else f"{x[0]}" for x in result])
+    return result
+
+
+def get_random_list_elem(traget_list: list):
+    """Helper method that randomly chooses an element from a list. numpy.random.choice() doesn't 
+    like to take elements from list()'s of tuples so this is the work around."""
+    return traget_list[np.random.randint(len(traget_list))]
```
Comment (chrisjurich) [inline enzy_htp/core/general.py:72]: Use np.random.choice() and add a `seed` argument

File: enzy_htp/core/general.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
+"""This module contains helper functions related to general python/python library features
```
Comment (chrisjurich) [inline enzy_htp/core/general.py:1]: This whole file needs better doc strings

**Review by chrisjurich — APPROVED** [review]
Looks really good. Documentation could be improved but we will take care of that in a later PR.

File: enzy_htp/chemical/atoms.py (hunk 1 above)
Comment (shaoqx) [reply enzy_htp/chemical/atoms.py:12]: Yapf will make it into one line if so. I want it to stay what it is rn. I searched about it and this is the solution they give https://github.com/google/yapf/issues/359

File: enzy_htp/core/general.py (hunk 2 above)
Comment (shaoqx) [reply enzy_htp/core/general.py:72]: I think seeding is a global behavior. That's the reason I didn't add `seed` for all random related functions.
I will change the np function

**Comment by shaoqx** [comment]
Cool! I revised those based on the comment!

**Pull request merged**